File: frontend/src/features/market-demand/countries.ts
export interface MarketDemandCountry {
  label: string;
  value: string;
}

export const DEFAULT_MARKET_DEMAND_COUNTRY = "USA";

export const MARKET_DEMAND_COUNTRIES: MarketDemandCountry[] = [
  { label: "United States", value: "USA" },
  { label: "China", value: "China" },
  { label: "United Kingdom", value: "UK" },
  { label: "Canada", value: "Canada" },
  { label: "Australia", value: "Australia" },
  { label: "Singapore", value: "Singapore" },
];

export function getMarketDemandCountryLabel(value: string | null | undefined): string {
  return (
    MARKET_DEMAND_COUNTRIES.find((country) => country.value === value)?.label ??
    "United States"
  );
}

File: backend/app/modules/marketdemandanalyst/countries.py
from dataclasses import dataclass


@dataclass(frozen=True)
class JobSearchCountry:
    label: str
    jobspy_country: str
    location: str


DEFAULT_JOB_SEARCH_COUNTRY = "USA"

SUPPORTED_JOB_SEARCH_COUNTRIES: dict[str, JobSearchCountry] = {
    "USA": JobSearchCountry(
        label="United States",
        jobspy_country="USA",
        location="United States",
    ),
    "China": JobSearchCountry(
        label="China",
        jobspy_country="China",
        location="China",
    ),
    "UK": JobSearchCountry(
        label="United Kingdom",
        jobspy_country="UK",
        location="United Kingdom",
    ),
    "Canada": JobSearchCountry(
        label="Canada",
        jobspy_country="Canada",
        location="Canada",
    ),
    "Australia": JobSearchCountry(
        label="Australia",
        jobspy_country="Australia",
        location="Australia",
    ),
    "Singapore": JobSearchCountry(
        label="Singapore",
        jobspy_country="Singapore",
        location="Singapore",
    ),
}

COUNTRY_ALIASES: dict[str, str] = {
    "": DEFAULT_JOB_SEARCH_COUNTRY,
    "us": "USA",
    "u.s.": "USA",
    "u.s.a.": "USA",
    "usa": "USA",
    "united states": "USA",
    "united states of america": "USA",
    "china": "China",
    "prc": "China",
    "uk": "UK",
    "u.k.": "UK",
    "united kingdom": "UK",
    "great britain": "UK",
    "canada": "Canada",
    "australia": "Australia",
    "singapore": "Singapore",
}


def normalize_job_search_country(raw_country: str | None) -> JobSearchCountry:
    normalized = (raw_country or "").strip().casefold()
    canonical = COUNTRY_ALIASES.get(normalized)
    if canonical is None:
        supported = ", ".join(
            country.label for country in SUPPORTED_JOB_SEARCH_COUNTRIES.values()
        )
        raise ValueError(
            f"Unsupported job search country: {raw_country!r}. "
            f"Supported countries: {supported}"
        )
    return SUPPORTED_JOB_SEARCH_COUNTRIES[canonical]
